lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -667,61 +667,76 @@ impl<S: Read + Write> RdpClient<S> {
         }
     }
 
+    fn write_rdpdr(&mut self, messages: Messages) -> RdpResult<()> {
+        let chan = &rdpdr::CHANNEL_NAME.to_string();
+        for message in messages {
+            self.mcs.write(chan, message)?;
+        }
+        Ok(())
+    }
+
     pub fn write_client_device_list_announce(
         &mut self,
         req: rdpdr::ClientDeviceListAnnounce,
     ) -> RdpResult<()> {
-        self.rdpdr
-            .write_client_device_list_announce(req, &mut self.mcs)
+        let messages = self.rdpdr.write_client_device_list_announce(req)?;
```

Comment:
I don't like that a method named `write_client_device_list_announce` doesn't actually write anymore. Can we rename to something like `handle_client_device_list_announce` instead?

Reply:
22a485a7c87fd525aeed6fea224c9a59b2fa2990